pkg/extension/sumologicextension/extension.go

```diff
@@ -124,7 +124,8 @@ func (se *SumologicExtension) storeCollectorCredentials() error {
 	if err := ensureStoreCredentialsDir(se.conf.CollectorCredentialsPath); err != nil {
 		return err
 	}
-	filenameHash, err := hash(se.conf.CollectorName)
+	key := se.conf.CollectorName + se.conf.Credentials.AccessID + se.conf.Credentials.AccessKey
```

Comment:
Shouldn't the `storeCollectorCredentials` and `ensureStoreCredentialsDir` functions be part of the `credsGetter`?

This would also make it easier to encapsulate the `CollectorName + AccessID + AccessKey` hash creation, which is repeated here and in `credsGetter.GetStoredCredentials` function.

Reply:
Probably a better name than `credsGetter` would then be `credsStore` because it would serve for both retrieving and storing the credentials.

Reply:
I like `credentialsStore` and I agree that it might be more obvious what it does granted we also have a method to store the credentials on it (not only to get them). 👍 

As for repeated hashing definition: I'd even go as far as make a function that would take the config and return the hashed string. This way if we ever decide to change the implementation this will be the only place to change. 👍 

@pmatyjasek-sumo WDYT?

As for other reorganization ideas I'm in for it but I'd suggest to skip the discussion on those in this PR

Reply:
> make a function that would take the config and return the hashed string
💯

Reply:
> Probably a better name than `credsGetter` would then be `credsStore` because it would serve for both retrieving and storing the credentials.

Thats why those `storeCollectorCredentials` and `ensureStoreCredentialsDir` are not parts of `credsGetter`. It grabs the authentication data for collector in two ways. If collector wasn't previously registered (file does not exists on filesystem) it register collector through the API and "gets" credentials. If file exists it also "gets" credentials but in this scenario it is taken from file on filesystem. Also both methods requires the same data as `AccessID`, `AccessKey` and `CollectorName` which are stored in config. `storeCollectorCredentials` needs some additional data like `collector credentials`